File: Core/Ecommerce.Domain/Entities/Product.cs
using Ecommerce.Domain.Common;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
// ReSharper disable EntityFramework.ModelValidation.UnlimitedStringLength

namespace Ecommerce.Domain.Entities
{
	public class Product : AuditableEntity
	{
		public int Id { get; set; }
		
		public int CategoryId { get; set; }
		
		[MaxLength(255)]
		public string Name { get; set; } = null!;
		
		public string Description { get; set; } = null!;
		
		public double Price { get; set; }
		
		public decimal AverageRating { get; set; }
		
		public int QuantityAvailable { get; set; }
		

		public string ImageUrl { get; set; } = null!;
		
		public IEnumerable<Review> CustomerReviews { get; set; } = null!;
	}
}

File: UI/ecommerce-ui/src/models/Product.ts
import {Review} from "./Review.ts";

export class Product {
    id: number;
    categoryId: number;
    name: string;
    description: string;
    price: number;
    averageRating: number;
    quantityAvailable: number;
    imageUrl: string;
    customerReviews: Review[];

    constructor(
        id: number,
        categoryId: number,
        name: string,
        description: string,
        price: number,
        averageRating: number,
        quantityAvailable: number,
        imageUrl: string,
        reviews: Review[] = []
    ) {
        this.id = id;
        this.categoryId = categoryId;
        this.name = name;
        this.description = description;
        this.price = price;
        this.averageRating = averageRating;
        this.quantityAvailable = quantityAvailable;
        this.imageUrl = imageUrl;
        this.customerReviews = reviews;
    }
}
